meson.build

```diff
@@ -223,6 +223,39 @@ if fc_id == 'gcc'
   endif
 endif
 
+proj_default_library = get_option('default_library')
+if proj_default_library == 'static'
+  is_static_build = true
+  message('MODFLOW 6 will be statically linked')
+else
+  is_static_build = false
+endif
+if is_static_build
+  if is_extended_build or is_parallel_build or is_netcdf_build
+    if is_static_build
+      is_static_build = false
+      message('Resetting static_build option for extended build')
```

Comment:
```diff
-message('Resetting static_build option for extended build')
+message('default_library=static not supported for extended, configuring shared build instead')
```

or "default_library=static only supported for standard builds, ..."
or "static builds not supported for netcdf/parallel/extended, ..."
or some way to say why static is reset/overridden and with what